src/pages/Quiz/index.tsx

```diff
@@ -1,6 +1,103 @@
-import React from "react";
+import axios from "axios";
+import { useQuizData } from "context";
+import React, { useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { Options } from "./components";
 import "./quiz.css";
 
 export function Quiz() {
-  return <section>Quiz Page</section>;
+  const { quizId, quesNo } = useParams();
+  const navigate = useNavigate();
+  const {
+    currentQuiz,
+    setCurrentQuiz,
+    currentQuizQuestions,
+    setCurrentQuizQuestions,
+    score,
+    setScore,
+  } = useQuizData();
+  const [ques, setQues] = useState(undefined);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(`/api/quizzes/${quizId}`);
+        setCurrentQuiz(response.data.quiz);
+        setCurrentQuizQuestions(
+          response.data.quiz.questions.map((i) => ({ ...i, isSelected: "" }))
+        );
+      } catch (error) {
+        console.log(error);
+      }
+    })();
+  }, [quizId, setCurrentQuiz, setCurrentQuizQuestions]);
+
+  useEffect(() => {
+    currentQuizQuestions && setQues(currentQuizQuestions[quesNo]);
+  }, [quesNo, setQues, currentQuizQuestions]);
+
+  const handleOptionChange = (option) => {
+    setCurrentQuizQuestions(
+      currentQuizQuestions.map((q) => {
+        if (q._id === ques._id) {
+          return { ...q, isSelected: option };
+        }
+        return { ...q };
+      })
```

Comment:
suggestion - this can be converted to single line using `? : ` operator.

Reply:
In next PR might be doing some more operations thats why did it like this.